sematic/api/endpoints/tests/test_runs.py

```diff
@@ -722,3 +728,17 @@ def test_get_run_external_resources(
 
     assert len(payload) == 1
     assert payload[0]["id"] == persisted_external_resource.id
+
+
+def test_set_run_user(persisted_user: User, run: Run, mock_requests):  # noqa: F811
```

Comment:
If you don't explicitly inject either the [`empty_settings_file`](https://github.com/sematic-ai/sematic/blob/main/sematic/config/tests/test_settings.py#L168-L171) or the [`no_settings_file`](https://github.com/sematic-ai/sematic/blob/main/sematic/config/tests/test_settings.py#L212-L215) fixture, this test will use your local settings file, and will behave differently based on the combination of its contents and of the current Sematic version.

Please standardize the behavior by adding one of the fixtures, and extract them to an importable fixture module.

Reply:
The code below overrides whatever is in those files so 🤷 .

Reply:
Developers running these tests might run into unrelated bugs caused by other contents in their settings file. I've had this happen before to me with such a test.

Although this would be an interesting way of discovering bugs, the behavior of the unit test would be non-deterministic, and distract the developer from ongoing work to investigate unrelated issues that are blocking them. This is frustrating.